### PR TITLE
MAINT-26695: No result for searched files

### PR DESCRIPTION
**Pull request by AzmiTouil**
The problem is caused by the special characters in title. To fix this problem, we have to escape these characters and add quotes around the text.

https://discuss.elastic.co/t/elasticsearch-escape-character/163216/1

### CODE REVIEW

**Comment by boubaker** [comment]
@AzmiTouil , could you please give a clear scenario to reproduce with exact filename that you would like to search.
In fact, I can't reproduce this with simple names.

**Comment by AzmiTouil** [comment]
> @AzmiTouil , could you please give a clear scenario to reproduce with exact filename that you would like to search.
> In fact, I can't reproduce this with simple names.

Yes, to reproduce it you just have to upload a file with a name containing special characters(ex : files@@@sample.doc), then type the full name( files@@@sample ) in the unified search => KO
NB : the file found when  typing the words "file" or "sample"

File: commons-search/src/main/java/org/exoplatform/commons/search/es/ElasticSearchServiceConnector.java [hunk 1]
```diff
@@ -193,7 +193,7 @@ protected String buildFilteredQuery(String query, Collection<String> sites, List
     if (StringUtils.isNotBlank(query)) {
       List<String> queryParts = Arrays.asList(query.split(" "));
```
Comment (boubaker) [inline commons-search/src/main/java/org/exoplatform/commons/search/es/ElasticSearchServiceConnector.java]: By applying this change :
https://github.com/Meeds-io/commons/blob/e98dc794a0e89bb688ec0c00c9171169307dd3ca/commons-search/src/main/java/org/exoplatform/commons/search/es/ElasticSearchServiceConnector.java#L196

The fuzzy search will not be applied anymore.

I suggest to change this line:
https://github.com/Meeds-io/commons/blob/e98dc794a0e89bb688ec0c00c9171169307dd3ca/commons-search/src/main/java/org/exoplatform/commons/search/es/ElasticSearchServiceConnector.java#L194
into : 
`
      List<String> queryParts = Arrays.asList(query.split("\\+\\-=\\&\\|><\\!\\(\\)\\{\\}\\[\\]\\^\"\\*\\?:\\/ @"));
`
This way you will delete special characters from searched words.
PS: we can't apply unfortunately use `whitespace` `analyzer` on query search (See https://www.elastic.co/guide/en/elasticsearch/reference/5.6/query-dsl-query-string-query.html)

Comment (AzmiTouil) [reply commons-search/src/main/java/org/exoplatform/commons/search/es/ElasticSearchServiceConnector.java]: Done ! Thanks

**Review by boubaker — APPROVED** [review]

**Review by ahamdi — APPROVED** [review]

**Pull request merged**